lambdas/postgres-migration-count-tool/iam.tf

```diff
@@ -0,0 +1,125 @@
+data "aws_iam_policy_document" "migration_count_assume_role_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+
+resource "aws_security_group" "postgres_migration_count" {
+  count = length(var.lambda_subnet_ids) == 0 ? 0 : 1
+
+  name   = "${var.prefix}-postgres-migration-count"
+  vpc_id = var.vpc_id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = var.tags
+}
+
+# migration_count_role
+resource "aws_iam_role" "postgres_migration_count_role" {
+  name                 = "${var.prefix}-postgres_migration_count"
+  assume_role_policy   = data.aws_iam_policy_document.migration_count_assume_role_policy.json
+  permissions_boundary = var.permissions_boundary_arn
+  tags                 = var.tags
+}
+
+data "aws_iam_policy_document" "postgres_migration_count_policy2" {
+  statement {
+    actions = [
+      "ecr:*"
+    ]
+    resources = ["*"]
+  }
+  statement {
+    actions = [
+      "ec2:CreateNetworkInterface",
+      "dynamodb:ListTables",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "lambda:GetFunction",
+      "lambda:invokeFunction",
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:DescribeLogStreams",
+      "logs:PutLogEvents",
+      "s3:ListAllMyBuckets",
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "s3:GetAccelerateConfiguration",
+      "s3:GetLifecycleConfiguration",
+      "s3:GetReplicationConfiguration",
+      "s3:GetBucket*",
+      "s3:PutAccelerateConfiguration",
+      "s3:PutLifecycleConfiguration",
+      "s3:PutReplicationConfiguration",
+      "s3:PutBucket*",
```

Comment:
Are all of these necessary?

Reply:
As we discussed, these need tidied up:

c53e507
43d2dc8